Scripts/Python/QueryGroupManagementOmeREDFISH.py

```diff
@@ -0,0 +1,706 @@
+            for i in data.items():
+                print("%s: %s" % (i[0], i[1]))
+        else:
+            print("\n- FAIL, GET request failed, status code %s returned, detailed error results: %s" % (response.status_code, data))
+            sys.exit()
+        
+
+
+def get_group_query_info():
+    response = requests.get('https://%s/api/GroupService/Groups(%s)/GroupQuery' % (ome_ip, args["q"]),verify=False,auth=(ome_username,ome_password))
+    data = response.json()
+    if response.status_code == 200 or response.status_code == 202:
+        pass
+    else:
+        print("- FAIL, GET request failed, status code %s returned, detailed error results: \n%s" % (response.status_code, data))
+        sys.exit()
+    if data["value"] == []:
+        print("\n- WARNING, no query information for OME group %s" % args["q"])
+        sys.exit()
+    print("\n- WARNING, getting query information for OME group %s -\n" % args["q"])
+    for i in data['value']:
+        for ii in i.items():
+            print("%s: %s" % (ii[0], ii[1]))
+        print("\n")
+
+def get_group_devices():
+    response = requests.get('https://%s/api/GroupService/Groups(%s)/Devices' % (ome_ip, args["d"]),verify=False,auth=(ome_username,ome_password))
+    data = response.json()
+    if data["value"] == []:
+        print("\n- WARNING, no devices detected for group ID %s" % args["d"])
+        sys.exit()
+    else:
+        print("\n- Device(s) detected for OME group %s -\n" % args["d"])
+        for i in data["value"]:
+            for ii in i.items():
+                print("%s: %s" % (ii[0], ii[1]))
+            print("\n")
+    
+                        
+
+def create_update_group(ID):
+    response = requests.get('https://%s/api/QuerySupportService/QueryContextSummaries' % (ome_ip),verify=False,auth=(ome_username,ome_password))
+    data = response.json()
+    context_id = ""
+    for i in data["value"]:
+        if "Devices" in i.values():
+            context_id = i['Id']
+    if context_id == "":
+        print("- FAIL, unable to locate context device ID")
+        sys.exit()
+    else:
+        pass
+    response = requests.get('https://%s/api/QuerySupportService/QueryContexts(%s)' % (ome_ip, context_id),verify=False,auth=(ome_username,ome_password))
+    if response.status_code == 200 or response.status_code == 202:
+        pass
+    else:
+        print("- FAIL, GET request failed to get query group field ID info, status code %s returned" % response.status_code)
+        sys.exit()
+    data = response.json()
+    field_id_list = []
+    for i in data['Fields']:
+        if i['FieldTypeId'] == 2:
+            field_id_list.append(str(i['Id']))
+    if args["fid"] and args["oid"] and args["value"]:
+        payload = {"GroupModel":{"Id": ID,"Name": args["name"],"Description": args["description"],"GlobalStatus": 0,"DefinitionId": 0,"MembershipTypeId":24,"ParentId": 1022},"GroupModelExtension" :{"FilterId":0,"ContextId":int(context_id),"Conditions":[]}}
```

Comment:
While 1022 may be a fixed number now across our instances for the ID of Query Groups it may be cleaner to programmatically derive this value - GET on GroupService/Groups using a filter - $filter=Name eq 'Query Groups' and then parsing the ID from the returned group